### ISSUE TITLE
Authentication failed with message "FragmentManager is already executing transactions"

### ISSUE THREAD

**Issue by filipleifer** [smstuebe/xamarin-fingerprint#198]
When Fingerprint plugin is not used on first page pushed into Navigation (NextPage in Demo) then you will get unexpected result if you try to await authentication result. If you use it on first page which is pushed into navigation (MainPage in Demo) the behaviour is correct. Demo app:

[FingerPrintBugDemo.zip](https://github.com/smstuebe/xamarin-fingerprint/files/5615816/FingerPrintBugDemo.zip)

## Steps to reproduce

1. Run demo app in Debug on Android device, set breakpoint to line 29 of NextPage.xaml.cs
2. Click on button "Click" on MainPage.
3. You get false authentication result with message "FragmentManager is already executing transactions" but fingeprint dialog is diplayed and awaiting your input in line 29 of NextPage.xaml.cs

## Expected behavior
You get false authentication result without wait on real result with message "FragmentManager is already executing transactions" but fingeprint dialog is still diplayed and awaiting your input.

## Actual behavior
You will be awaiting the real fingerprint dialog result.

## Configuration

**Version of the Plugin:** Latest 2.1.2

**Platform:** Android 8.0 / Android 9.0

**Device:** Huawei P20 Lite ANE-LX1 / Sony Xperia XZ1 G8342

**Comment by smsissuechecker** [comment]
Hi @filipleifer,

I'm the friendly issue checker.
Thanks for using the issue template :star2:
I appreciate it very much. I'm sure, the maintainers of this repository will answer, soon.

**Comment by rj-thomas** [comment]
Also experiencing this issue on Android 10, using latest 2.1.3

**Comment by kgbuddhima** [comment]
Yes, I too having the same issue on Android Q. 

<img width="1052" alt="Screenshot 2021-02-02 at 13 23 52" src="https://user-images.githubusercontent.com/14989528/106568970-01683280-655a-11eb-94e8-7c8a4649913a.png">

**Comment by wstanczewski** [comment]
I'm having the same issue, Android 11

**Comment by wstanczewski** [comment]
I think this issue is a duplicate of #182

**Comment by wstanczewski** [comment]
Using this plugin like that:
`var result = await Device.InvokeOnMainThreadAsync(() => CrossFingerprint.Current.AuthenticateAsync(new("","")))`
solves the issue :)

**Comment by Ulrizza** [comment]
I have the same issue and @wstanczewski's solution is not working for me.  
As a workaround I wait 500ms before calling `AuthenticateAsync` and it's working.  
```
await Task.Delay(500);
var result = await Plugin.Fingerprint.CrossFingerprint.Current.AuthenticateAsync([...]);
```

**Comment by maheshkumar1989** [comment]
> Using this plugin like that: `var result = await Device.InvokeOnMainThreadAsync(() => CrossFingerprint.Current.AuthenticateAsync(new("","")))` solves the issue :)

We tried this process, it worked with debug mode but it's not working in the release mode, Authentication process is not invoked. If any one has any suggestion, any help is appreciated

**Comment by IDispose** [comment]
> I have the same issue and @wstanczewski's solution is not working for me. As a workaround I wait 500ms before calling `AuthenticateAsync` and it's working.
> 
> ```
> await Task.Delay(500);
> var result = await Plugin.Fingerprint.CrossFingerprint.Current.AuthenticateAsync([...]);
> ```

This is the only solution that worked. Observed that stepping thru code and waiting for a second in the debugger doesn't throw the error. Hacky, but works.

**Comment by Francescolis** [comment]
I had the same issue, and this code fixed it :
``
await Task.Yield();
``
``
FingerprintAuthenticationResult result = await _fingerprint.AuthenticateAsync(request);
``
